### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by ltratt**
This PR adds more tests to snare. While there is always more that we could do, this feels like it gets us to a "minimum acceptable" level of testing. Better late than never!

### CODE REVIEW

File: tests/queue.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
+use std::{error::Error, fs::read_dir, thread::sleep};
+use tempfile::Builder;
+
+mod common;
+use common::run_success;
+
+// Note that `sleep_ms` has to be a fairly high value as we really hope that snare has finished
```
Comment (vext01) [inline tests/queue.rs:7]: sleep_ms -> sleep_s?

Comment (ltratt) [reply tests/queue.rs]: Oops! Fixed in 9d08540.

**Comment by vext01** [comment]
Please squash

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**